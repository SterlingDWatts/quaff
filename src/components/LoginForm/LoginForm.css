.LoginForm {
  margin: auto;
}

.LoginForm h1 {
  font-size: 32px;
  font-weight: 600;
  text-align: center;
  line-height: 1;
}

.LoginForm .hint {
  margin: 0px 0px 32px;
  text-align: center;
  line-height: 1;
  font-size: 13px;
  color: #808080;
}

.LoginForm__buttons {
  max-width: 288px;
  margin: 0px auto;
}

.LoginForm__other_links {
  font-size: 16px;
  margin-top: 16px;
  text-align: center;
  color: grey;
}

.LoginForm__other_links a {
  color: #2d8cfb;
  font-weight: 600;
}

.LoginForm--error {
  margin-bottom: 8px;
}

.LoginForm--error p {
  font-size: 16px;
  color: rgb(253, 151, 38);
  line-height: 1;
  font-weight: 600;
}
