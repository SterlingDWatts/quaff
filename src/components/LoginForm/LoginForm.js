import React, { Component } from "react";
import { Link } from "react-router-dom";
import {
  LabelGroup,
  Label,
  Input,
  Button,
  Required,
  ValidationError,
} from "../Utils/Utils";
import AuthApiService from "../../services/auth-api-service";
import ValidationService from "../../services/validation-service";
import ModulesApiService from "../../services/modules-api-service";
import "./LoginForm.css";

class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {},
  };

  state = {
    error: null,
    username: {
      value: "",
      touched: false,
    },
    password: {
      value: "",
      touched: false,
    },
  };

  handleUsernameChange = (username) => {
    this.setState({
      username: {
        value: username,
        touched: true,
      },
    });
  };

  handlePasswordChange = (password) => {
    this.setState({
      password: {
        value: password,
        touched: true,
      },
    });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { username, password } = e.target;
    this.setState({
      error: null,
    });

    AuthApiService.postLogin({
      username: username.value,
      password: password.value,
    })
      .then((res) => {
        this.setState({
          username: { touched: false, value: "" },
          password: { touched: false, value: "" },
        });
        this.props.onLoginSuccess();
        if (window.localStorage.getItem("test")) {
          const test = JSON.parse(window.localStorage.getItem("test"));
          ModulesApiService.insertTest(test.id, test.views, test.score)
            .then((quiz) => {
              window.localStorage.removeItem("test");
            })
            .catch();
        }
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error, username, password } = this.state;
    const usernameError = ValidationService.validateUsername(username.value);
    const usernameLabelError = ValidationService.validateUsernameLabel(
      username.value,
      username.touched
    );
    const passwordError = ValidationService.validatePassword(password.value);
    const passwordLabelError = ValidationService.validatePasswordLabel(
      password.value,
      password.touched
    );
    return (
      <form className="LoginForm" onSubmit={(e) => this.handleSubmit(e)}>
        <header>
          <h1>Login</h1>
        </header>
        <div className="hint">
          <Required /> required fields
        </div>
        <LabelGroup className="LoginForm__username">
          <Label htmlFor="username">
            {usernameLabelError}Username <Required />
          </Label>
          <Input
            error={usernameLabelError}
            name="username"
            id="username"
            type="text"
            onChange={(e) => this.handleUsernameChange(e.target.value)}
          />
          <ValidationError message={usernameError} touched={username.touched} />
        </LabelGroup>
        <LabelGroup className="LoginForm__password">
          <Label htmlFor="password">
            {passwordLabelError}Password <Required />
          </Label>
          <Input
            error={passwordLabelError}
            name="password"
            id="password"
            type="password"
            onChange={(e) => this.handlePasswordChange(e.target.value)}
          />
          <ValidationError message={passwordError} touched={password.touched} />
        </LabelGroup>
        <div className="LoginForm--error">
          <ValidationError message={error} touched={true} />
        </div>
        <div className="LoginForm__buttons">
          <Button type="submit" className="form__button">
            Login
          </Button>
        </div>
        <div className="LoginForm__other_links">
          Don't have an account?{" "}
          <Link to="/create-account">Create Account</Link>
        </div>
      </form>
    );
  }
}

export default LoginForm;
