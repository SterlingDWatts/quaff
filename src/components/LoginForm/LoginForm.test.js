import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";
import LoginForm from "./LoginForm";

describe("LoginForm Comoponent", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <LoginForm />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const wrapper = shallow(<LoginForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("has a disabled submit button when rendered", () => {
    const wrapper = mount(
      <BrowserRouter>
        <LoginForm />
      </BrowserRouter>
    );
    const button = wrapper.find("button");
    const disabled = button.prop("disabled");
    expect(disabled).toBe("Username is required");
  });

  it("enables the submit button when proper information is added", () => {
    const wrapper = mount(
      <BrowserRouter>
        <LoginForm />
      </BrowserRouter>
    );
    wrapper.find("input#username").simulate("change", { target: { name: "username", value: "RobertFHinkel" } });
    wrapper.find("input#password").simulate("change", { target: { name: "password", value: "aaAA11!!" } });
    const button = wrapper.find("button");
    const disabled = button.prop("disabled");
    expect(!!disabled).toBe(false);
  });
});
