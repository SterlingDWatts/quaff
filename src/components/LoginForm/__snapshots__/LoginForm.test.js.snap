// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Comoponent renders the UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <header>
    <h1>
      Login
    </h1>
  </header>
  <div
    className="hint"
  >
    <span
      className="Required "
    >
      *
    </span>
     required fields
  </div>
  <div
    className="LabelGroup LoginForm__username"
  >
    <label
      className="Label "
      htmlFor="username"
    >
      Username 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="username"
      name="username"
      onChange={[Function]}
      type="text"
    />
    <div
      className="NoError"
    >
       
    </div>
  </div>
  <div
    className="LabelGroup LoginForm__password"
  >
    <label
      className="Label "
      htmlFor="password"
    >
      Password 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="password"
      name="password"
      onChange={[Function]}
      type="password"
    />
    <div
      className="NoError"
    >
       
    </div>
  </div>
  <div
    className="LoginForm--error"
  >
    <div
      className="NoError"
    >
       
    </div>
  </div>
  <div
    className="LoginForm__buttons"
  >
    <button
      className="Button form__button"
      type="submit"
    >
      Login
    </button>
  </div>
  <div
    className="LoginForm__other_links"
  >
    Don't have an account yet?
     
    <a
      href="/create-account"
      onClick={[Function]}
    >
      Create Account
    </a>
  </div>
</form>
`;
